src/coreclr/jit/jitconfigvalues.h

```diff
@@ -594,6 +594,8 @@ OPT_CONFIG_INTEGER(JitDoIfConversion, "JitDoIfConversion", 1)
 OPT_CONFIG_INTEGER(JitDoOptimizeMaskConversions, "JitDoOptimizeMaskConversions", 1) // Perform optimization of mask
                                                                                     // conversions
 
+RELEASE_CONFIG_INTEGER(JitOptimizeAwait, "JitOptimizeAwait", 1) // Perform optimization of Await intrinsics
```

Comment:
I am assuming this will be OFF by default?

Reply:
This is ON by default. This is what allows to bypass materialization of `Task` when one Async method awaits another. In production it is supposed to be always on.

The optimization is in theory optional though, and disabling it has revealed bugs in the past. It is also useful to measure the performance impact while developing.

The plan is to eventually remove the knob or make it Checked.
Tracked in https://github.com/dotnet/runtimelab/issues/3012  

(probably need to port the issue to the main repo once code moves, perhaps applies to other Async issues in runtime lab as well)

Reply:
If potential regressing of existing scenarios is a concern - this knob/optimization will only have effect inside Async methods. If not Async methods, then no effect.

```c
                    if (compIsAsync() && JitConfig.JitOptimizeAwait())
                    {
                        isAwait = impMatchAwaitPattern(codeAddr, codeEndp, &configVal);
                    }

                    . . . 
```